### ISSUE TITLE
Drones randomly falls onto the ground

### ISSUE THREAD

**Issue by pPatrickK** [USC-ACTLab/crazyswarm#64]
Sometime a drone randomly falls down.
We see this problem since the new release. It seems like the engines just shut off for no reason.
Do you have some hints for us to solve this problem?
We saw this problem with different scripts. For example, "figure8_csv.py" and others. Even on own scripts with simple goto commands.
We see those "Don't know ack" and one "link quality low" error messages:

> [ INFO] [1526389834.069563575]: [cf20] Update parameters
> [ INFO] [1526389834.131466331]: Update params: 0.072950 s
> [ INFO] [1526389834.131564272]: Started 1 threads
> [ INFO] [1526389839.818718904]: [cf20] Upload trajectory
> [ WARN] [1526389839.842813651]: Latency too high! Is 0.009258 s.
> [ WARN] [1526389839.956615973]: Latency too high! Is 0.012839 s.
> [ WARN] [1526389839.992218644]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389839.993758488]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389839.998673977]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.001823660]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.003410951]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.005012624]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.008210921]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.009796982]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.013099888]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.014687268]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.019393615]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.022582420]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.024152848]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.028920479]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.032163458]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.033717457]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.036857618]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.038421661]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.040019073]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.043349168]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.044896328]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.047966159]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.049552269]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.052748658]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.054290630]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.057396114]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.058900243]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.062170358]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.063746538]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.066887262]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.068442953]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.070165117]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.073449530]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.075050056]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.078349725]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.089618555]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.090293804]: Latency too high! Is 0.011876 s.
> [ WARN] [1526389840.092082812]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.094823300]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.097462835]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.098389554]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.099342653]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.101932494]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.102870241]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.103765166]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.106355384]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.107907640]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.112907071]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.114475663]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.117611540]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.119152273]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.123952395]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.127157452]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.128264511]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.129214621]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.132868530]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.133833855]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.136345869]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.137321734]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.138289976]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.139257685]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.141888828]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.142815280]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.144378095]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.147521364]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.149016243]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.151694406]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.152648228]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.154381757]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.157178942]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.158148632]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.159115167]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.161634030]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.162610507]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.164374449]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.167577477]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.169149675]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.171864246]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.172832428]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.174753110]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.178538107]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.181043313]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.181958203]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.182914033]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.186423969]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.187962466]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.189556105]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.192554753]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.193525260]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.196211985]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.197131617]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.199024276]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.201546816]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.202531071]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.216440609]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.219221229]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.221852824]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.227767528]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.229641384]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.232231703]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.233161759]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.234140043]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.237552069]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.238512122]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.243004950]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ INFO] [1526389840.243083594]: [cf20] Uploaded trajectory
> [ INFO] [1526389840.249163031]: Takeoff!
> [ WARN] [1526389840.254750272]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.266567522]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.278135583]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.289666666]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.300958504]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.358705486]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.370589663]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.382153924]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.393682078]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.405739716]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.452570268]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.464062728]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.487386359]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.569204406]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.580941749]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389840.592694891]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.639209001]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.651436748]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.662703402]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.674044974]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.685632835]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.697191212]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.732716882]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.755826923]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.767338911]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.778688138]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.790786710]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.802392523]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.826094866]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.849214096]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.860937356]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.883828541]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.895848566]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.907451424]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.918962713]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389840.930684338]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ INFO] [1526389842.785727153]: [cf20] GoTo
> [ WARN] [1526389844.058618797]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389844.070563164]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ INFO] [1526389845.301481971]: Start trajectory!
> [ WARN] [1526389846.446070010]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389846.457519444]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389846.468998116]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389846.480510021]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389846.491994605]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ INFO] [1526389854.656097615]: Land!
> [ WARN] [1526389856.447597556]: [cf20] Link Quality low (0.600000)
> [ INFO] [1526389860.526789694]: [cf20] Upload trajectory
> [ WARN] [1526389860.638087137]: Latency too high! Is 0.009887 s.
> [ WARN] [1526389860.685251234]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.685945679]: Latency too high! Is 0.012972 s.
> [ WARN] [1526389860.691822144]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.695021418]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.698259279]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.701282798]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.702832142]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.704437251]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.707536356]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.709096231]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.712448930]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.714021812]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.717331305]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.718902092]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.722059000]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.723565243]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.726758897]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.728313108]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.731568517]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.733110729]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.736288170]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.737840512]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.739385100]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.742390990]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.743961511]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.747303690]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.748878817]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.752031982]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.753583557]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.756798426]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.758295105]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.759819099]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.763015906]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.764541697]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.767634077]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.769136671]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.772280321]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.773822906]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.776887825]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.778390146]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.781561704]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.783088869]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.784638664]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.787982583]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.791124570]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.794572557]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.797748393]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.799683999]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.802479284]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.803444157]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.814682248]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389860.822800715]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.827819496]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.830525147]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.835808998]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.837517049]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.838454391]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.842071176]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.843725041]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.846151012]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.847088188]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.848018873]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.851592489]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.852494837]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.853421377]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.857605094]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.860928180]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.863741851]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.866477689]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.867427184]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.870905542]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.871854768]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.872848358]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.873811083]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.877054169]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.878999532]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.881668509]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.882640913]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.883561020]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.887696135]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.888688165]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.891132323]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.892692624]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.894405331]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.896988011]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.897966705]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.898961227]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.901478219]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.902429616]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.903401908]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.906171710]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.908491905]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.911148056]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.912118836]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.913101982]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.914074606]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.916695202]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.917631943]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.918549163]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.920939859]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ INFO] [1526389860.922699676]: [cf20] Uploaded trajectory
> [ INFO] [1526389860.931151489]: Takeoff!
> [ WARN] [1526389860.968816750]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.980558552]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389860.991940381]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389861.003463785]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389861.038377111]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389861.050745382]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389861.062365447]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389861.108872434]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389861.120608795]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389861.143676022]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389861.155346317]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389861.213909580]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389861.225491331]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389861.272111463]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389861.307002202]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389861.388903787]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389861.400336912]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389861.423208396]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389861.435452012]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389861.447092513]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389861.458589730]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389861.482167447]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389861.493696106]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389861.505518432]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389861.517031476]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389861.528598431]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389861.540178674]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389861.563277043]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389862.635539956]: [cf20] Link Quality low (0.660000)
> [ INFO] [1526389863.464142074]: [cf20] GoTo
> [ WARN] [1526389864.595110974]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ INFO] [1526389865.972966376]: Start trajectory!
> [ WARN] [1526389867.107768560]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389867.119214769]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389867.130961433]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389867.142836097]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389867.154276688]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389867.166046096]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ INFO] [1526389875.326129078]: Land!
> [ INFO] [1526389885.421468952]: [cf20] Upload trajectory
> [ WARN] [1526389885.449781689]: Latency too high! Is 0.009361 s.
> [ WARN] [1526389885.516215140]: Latency too high! Is 0.013129 s.
> [ WARN] [1526389885.538537062]: Latency too high! Is 0.015744 s.
> [ WARN] [1526389885.569587051]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.573053723]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.576389735]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.577912618]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.582577550]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.584096403]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.587431284]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.588962850]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.592173395]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.595296535]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.598674892]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.602042912]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.603560913]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.608417062]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.611619296]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.613212895]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.616322806]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.617872674]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.621098590]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.622607818]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.624161873]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.627522153]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.629087031]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.632381221]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.633924195]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.637196147]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.638765979]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.651116546]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.654230895]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.657321133]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.660648944]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.664023465]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.667307189]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.672119457]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.673608709]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.677009619]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.680119406]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.683264543]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.686356532]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.689121645]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.691574917]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.692495765]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.693439543]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.696091813]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.697004922]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389885.700724776]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.701708416]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.706745036]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.708268049]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.711333122]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.712879597]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.717694037]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.719235491]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.722472154]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.723975729]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.728158092]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.730609818]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.731582278]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.732537761]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.736096137]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.737066458]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.738075007]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.740532392]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.743037599]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.746117827]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.747690749]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.749221151]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.754047575]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.757293508]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.758872848]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.761411373]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.762325234]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.763285218]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.765838779]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.766866143]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.767836254]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.771516276]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.772413617]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.773354661]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.784433589]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.787614200]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.790718308]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.793386664]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.795946492]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.797722222]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.798664658]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.802030454]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.803034099]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.805599554]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.807919165]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.811016285]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.812570002]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.814100393]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389885.818877022]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.821577622]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.822557354]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.823519034]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.828059081]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.830710178]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.832272258]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ INFO] [1526389885.832330964]: [cf20] Uploaded trajectory
> [ INFO] [1526389885.840853728]: Takeoff!
> [ WARN] [1526389885.843833006]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.855527674]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.867088791]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.901737382]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.924707926]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.936334456]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.947861731]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389885.994156782]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389886.005745251]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389886.017226698]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389886.028741792]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389886.052045274]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389886.086786204]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389886.098246434]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389886.110276199]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389886.122009787]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389886.133498695]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389886.145149360]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389886.203162669]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389886.215439366]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389886.227089820]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389886.238635060]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389886.250565122]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389886.297072000]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389886.320147239]: Don't know ack: Port: 4 Channel: 2 Len: 7
> [ WARN] [1526389886.483256626]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389886.495326810]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389886.518482411]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389886.530168482]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389886.541567534]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389886.623797926]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389886.635239798]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389886.646636864]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389886.658150420]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389886.669962605]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389886.740040288]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389886.751419491]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389886.763022609]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389886.775148730]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389886.786696105]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389886.798108752]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389886.810092996]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389886.833107051]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389886.844931844]: Don't know ack: Port: 8 Channel: 0 Len: 5
> [ WARN] [1526389887.498618867]: [cf20] Link Quality low (0.620000)
> [ INFO] [1526389888.373566946]: [cf20] GoTo
> [ INFO] [1526389890.890836045]: Start trajectory!
> [ INFO] [1526389900.238371515]: Land!
>

**Comment by whoenig** [comment]
I am sorry about the "Don't know ack" warnings - they are harmless, and I created an issue in crazyflie_cpp to address that issue (see https://github.com/whoenig/crazyflie_cpp/issues/2). 

Did you disable logging? The link quality is actually not analyzed in swarm operation with logging disabled. If logging or any other direct drone<->PC communication is enabled (instead of broadcasts), the broadcasts and other operations with share the radio and you increase the risk of not getting enough broadcast messages about the position. If you have the LED-ring installed, it will show you the link quality by default (green: good; red: bad).

**Comment by pPatrickK** [comment]
Observation from today:

I've increased the timeout timeout time as suggested in [https://github.com/USC-ACTLab/crazyswarm/issues/63](url)

I flew just one drone with logging disabled,
randomly the drone just shut off and fell to the ground. This always happened when a new goto was send.

Any ideas?

We install today our realtime spectrum analyzer to monitor the traffic.

**Comment by pPatrickK** [comment]
Picture 1 shows the bar of channel 82 while the server is running.
Picture 2 shows channel 82 while the server and the flie is off. The channel seems pretty clean. 

We tested also the following:
-Pull out the dongle while the cf hovers -> cf drifts away
-Kill the server over the task manager -> cf drifts away
-If we shut down the server by ctrl+c -> cf shuts engines off as with our problem.

So at some point the cf must decide to cut of the engines for some reason. What would be the best way to debug this? 

![img_0242](https://user-images.githubusercontent.com/30862002/40116820-2f0238c0-5915-11e8-8a39-04f614bcb87b.JPG)
![img_0243](https://user-images.githubusercontent.com/30862002/40116822-319cb93e-5915-11e8-90b6-c28c750ce8c8.JPG)

**Comment by whoenig** [comment]
That's a neat setup - sounds like we can rule out communication issues. A few ideas:
1.) After the crash, is the CF is a failure mode (usually indicated by the red LED being consistently on)? This would indicate that there is an issue with the on-board computation.
2.) Did you verify that your goTo commands are feasible? In other words, what is the distance and desired time you send? I have seen such crashes if a) the distance is too long for the desired time, or b) the desired time is very short (<0.5s). In the first case the resulting trajectory is simply infeasible. In the second case I assume we run into some numerical issue (we only use single-precision computation on-board).
3.) If you can reproduce with a single drone, you can also re-enable logging and look at the setpoint and state information to see if there is a sudden jump (which would indicate a planning issue).

The ultimate/best way to debug the issue is to attach the uSD-card deck and log setpoint, state, and vicon information in every control loop iteration and analyze the data after the crash. We had that functionality in the old firmware, but not in the updated one (we are hoping to use the existing uSD-card infrastructure of the official firmware.) This basically helps you to also figure out which component (state estimation or control) failed first.

**Comment by pPatrickK** [comment]
After a crash the red LED is constantly on, yes.
We give the flie enough time for the movement. For example two seconds for one meter. With the last version, those values worked well. We ordered some uSD card decks to check your last point.
If we start logging over the plotter (state and estimated state for example), the drone crashes after a short time.

We attached some images of the plots. Red curves are target values and blue are estimated values. First image is x- second is y- and third is z-value.

![x](https://user-images.githubusercontent.com/30862002/40183547-690091fa-59ee-11e8-87f7-aacdc47920b2.png)

![y](https://user-images.githubusercontent.com/30862002/40183546-68e2e2e0-59ee-11e8-986a-3a1399964fca.png)

![z](https://user-images.githubusercontent.com/30862002/40183545-68c2368a-59ee-11e8-89f6-772326618207.png)

We also made a [video](https://youtu.be/Rl2rE8j0cEY) of the flight.

**Comment by whoenig** [comment]
The plots and scripts look good.

In your video, I noticed that the red light turns fully on after the first waypoint. The CF then continues one or two more waypoints, before falling down. While (or after) falling down, the CF seems to reboot (and according to your information the red light will be on.)

That the red light is on during the flight while still operating means low battery voltage. (If you have the LED ring installed, the ring will start blinking to indicate low battery.)
On the other hand, the falling and red light after the reboot likely means that there was a firmware issue. The CF has a watchdog that resets the drone if necessary, for example if there was a hardfault (e.g. division by 0), memory access fault, or some computation simply took too long.

Things to try out:
1. Make sure that your batteries are sufficiently charged. You can check using chooser.py before the flight.
2. After the drone crashed, try to connect with the CF client and look at the debug console (or, use the console tool in crazyflie_tools). It might provide you with some details on why the drone crashed.
3. If there is not much detail in 2, you can also try a firmware that is compiled with DEBUG=1. This will have two effects: a) the watchdog is disabled, so if it is just a matter of computation time there won't be a crash, and b) you get debug symbols. If you happen do have the debug adapter kit and a JTAG-debugger, you can attach with gdb after the crash and try to get a stacktrace using the `bt` command.

I might have some time tomorrow to double check the operation on our side with your script. I assume you use the latest crazyswarm master branch, together with the pre-compiled firmware we provide?

**Comment by pPatrickK** [comment]
Yes, we use the latest branch and firmware.
We use the motion capture tracking here with Nexus2.6.

We've made the config.mk with 
```
PLATFORM=CF2
DEBUG=1
CLOAD=1
```
This should bring us in debug or?
We currently getting the right connector.

In this [video](https://youtu.be/L6jrqH5coEY) you see that the cf reboots after it falls(which shouldn't be since we are in debug?). After the reboot the red light is flashing as in the [failed self test](https://www.bitcraze.io/getting-started-with-the-crazyflie-2-0/#infostep18).

Best regards.

**Comment by whoenig** [comment]
You are correct that the provided config.mk should create a DEBUG build where the CF should never reboot, unless power-cycled. Did you use `make clean` followed by `make`? The Makefile will not detect changes in the config.mk file.

I have not seen such reboots as in your video for a debug build, unless there was some mechanical problem (lose battery wires).

One problem can be related to flashing. If you use the official bitcraze tool, it doesn't use verification. I have seen on multiple instances that flashed firmwares were not identical to the firmware on the PC, resulting in all kinds of weird bugs. You should use the `flash` tool part of `crazyflie_tools` instead, or the JTAG debug adapter (with `make flash`) - both use verification by default. Note, that `flash` was not working in the last release, but I just pushed a fix for that yesterday.

I didn't get around testing your example script in our facility yet, but it's still high up on my ToDo list for next week.

**Comment by whoenig** [comment]
I just tested it on my side and I can reproduce the crashes with the crazyswarm firmware. However, as a workaround, it seems to be fine to use the official firmware:
1. `rosrun crazyflie_tools flash --target stm32 --filename ../prebuilt/cf2_bitcraze.bin`
2. Change [this](https://github.com/USC-ACTLab/crazyswarm/blob/master/ros_ws/src/crazyswarm/launch/hover_swarm.launch#L10) line to "bitcraze".

I will update the thread once I found (& fixed) the issue in the crazyswarm firmware. The crazyswarm firmware requires more stack because it has a more complex EKF, so my best guess is that there are some memory corruptions (since the goTo planning code is otherwise identical between the firmwares).

Note that the bitcraze firmware has the disadvantage that it only fuses position information, so you can't hover at a point forever (moving sideways allows the EKF to correct yaw drift).

**Comment by pPatrickK** [comment]
We are currently working with the commit from March 22, 2018 (50e81754da9264457ca5b0a7b8ef0710fbbae759). We can observe that with more drones the they get less stable up to a point where the first crashes start. With 20 drones we usually have 2-3 crashes. Right now we are assuming that the error stems from a firmware error. 
Observations: 
 
1. If I pause tracker for half a second and press play afterwards -> drone drifts off a bit and corrects then to the setpoint with high dynamic. If I wait longer the drone exceeds its dynamic limits and crashes.

2. During takeoff one or more of the drones performs a backflip, although all drones are having correct poses in tracker. These backflips sometimes also occur during normal flight. 

3. Another issue occurs if we use a small swarm in contrast to a single drone. One drone can fly highly dynamical and without issues, while e.g. with 10 drones all loose their stable pose during flight (fluctuations in rotation and position) and a few drones crash with a similar behavior as in 1. .

We are assuming two main error sources. First package loss/excessive channel latency during transmission. Second, bad estimations on the drone itself.

Do you have further suggestions on the next steps of the debugging process. We have 2 SD-Card Decks to attach to the drones and a ST-Link Debugger to our disposal. Although we are not sure, whether we are able to connect the ST-Link Debugger to a drone in motion, without influencing its flight behavior. In which directions should we investigate first with the debugging process?

Also is frequency hopping available for the NRF chip ? 
Thanks in advance!

**Comment by whoenig** [comment]
We used this particular commit to fly 13 drones without any issues (trajectory execution).

* Are you operating in waypoint following mode or trajectory execution? Unfortunately, we didn't test waypoint following extensively lately.
* You mention that it's correct in tracker. Are you using Tracker for tracking or libmotioncapture? I have seen such backflips only if using Vicon Tracker when axis-flips occurred. If you use Tracker, try to enable a motion filtering model.
* If I remember correctly, you have LED rings and a spectrum analyzer. If the analyzer doesn't show issues and the LED rings are green it should be fine. There is no HW support for frequency hoping in the NRF chip. It might be possible to implement that in SW though.

Debugging:
* For uSD-cards I recommend going to the latest official firmware (and a newer crazyswarm version). We just implemented proper support of logging of any onboard variables - see the wiki here: https://wiki.bitcraze.io/projects:crazyflie2:expansionboards:microsd. If you use the provided config.txt, you can look at the state estimate and see if there are any weird flips. You could also add variables for the received external state estimate to check if there are any latency issues.
* The debugger can't be used while flying. It is mostly useful for three purposes: a) flash firmware quickly (with verification), b) develop and step through firmware, c) figure out why a crash occurred. If you believe the firmware crashed (or rebooted) after you observed odd behavior, you can attach the debugger after the crash occurred and hopefully get an idea where it happened.

**Comment by whoenig** [comment]
Just as update for the waypoint following crashes (given your python script): I identified and fixed the issue in the crazyswarm firmware. At the same time I merged the latest official firmware master into our code base, so the crazyswarm firmware now also has proper uSD card logging support.

**Comment by pPatrickK** [comment]
Hi,
thanks, that worked.

Still we face a few crashes during a flight of the swarm. We logged some variables with the sd deck and think that we have a combination of issues which could be the reason for the crashes.

In our hall (build out of sheet metal) we have strong interference which results in a lot of spots where the packet rate is 0. 
Our drones differ in performance, picture 3 & 4. In picture 3 (drone that crashed) one can see that strong deviatito ons between the engines. When the dynamic gets to high the engine will run into the threshold of ~65000. (Picture 1&3 = drone1; 2&4 = drone2)  

Questions
1. Is the threshold ~65000 (variable: motor.m1-m4) really the physical maximum or is it limited by software? (is it possible the get more power out of the engines without changing the hardware?)
2. Have you experienced deviations between the hardware of the crazyflies (battery, engines...) ?
3. How should we interpret the CTRL yaw (first plot in pic 1&2,variable: ctrltarget.yaw)
4. Have you guys tweaked the drones in a way that the engine signals are similar to each other?
5. How could you place the markers underneath the rotor? (We had once placed a 1.5 mm rod underneath a rotor which already destroyed the aerodynamics, the drone could not take off)
6. Have you guys ever tried to run the swarm with 250K instead of 2Mbps, any experiences? With that we can achieve a more stable packet rate (varies strongly with location). Best is 0.92.

Any more ideas ?

![ctrl_target_vs _vicon](https://user-images.githubusercontent.com/30862002/41308257-bc071f1c-6e7b-11e8-8409-777bc86221d5.png)
![ctrl_target_vs _vicon7](https://user-images.githubusercontent.com/30862002/41308258-bc67f60c-6e7b-11e8-95d0-0d6aee005d8b.png)
![motor](https://user-images.githubusercontent.com/30862002/41308260-bc962cfc-6e7b-11e8-8e25-f7c5dfef395e.png)
![motor7](https://user-images.githubusercontent.com/30862002/41308261-bcc248b4-6e7b-11e8-8cd1-a703c968be7c.png)

**Comment by whoenig** [comment]
I don't recall looking at how close our thrust output to saturation is. I can check next time I run experiments. Which trajectory are you flying at which speed? What is the weight of your drone?
There are a few tricks to deal with saturation that we haven't implemented yet, but for "normal" speeds that shouldn't be a problem.

1. The thrust is unit-less and essentially the PWM-output mapped between 0 and 65536 (see https://github.com/bitcraze/crazyflie-firmware/blob/7ae809eeb5b48b5d84649c713accd2d7bf3df338/src/drivers/src/motors.c#L201-L232). The actual thrust depends on the battery voltage, so newer/more powerful batteries can produce more thrust. Note that bigger batteries can cover the antenna, which will make your reception worse, so stick to the ones that are roughly the original size. (We use [these](https://hobbyking.com/en_us/turnigy-nano-tech-260mah-1s-35-70c-lipo-pack-qr-ladybird-genius-cp-mini-cp-5pcs.html?wrh_pdp=7)
2. Definitely, especially after a few crashes. We frequently replace propellers in that case (hard to visually inspect actually), rarely motor mounts (only if visibly damaged), and maybe only once a motor. The I-parts of our controller take care of differences fairly well.
3. This is the yaw setpoint in degrees. I believe your jumps might come from "mode-switches" (e.g., if you take-off, land, or go to a new waypoint). Whenever the on-board planner is called, we plan from the current state to the new goal. You can a "jump" if the current state was not identical to the setpoint. We might change that behavior to plan from the current setpoint instead, to avoid any jumps in the setpoint.
4. No, we just let the I-part of our controller take care of such differences.
5. A while ago I did play around with 3D printed frames etc. for markers. Basically anything I tried had a significant effect on aerodynamics, even when not underneath. One thing you could try is to use active IR LEDs instead of big markers (similar to the VICON Active Wand). This should be easy with the expansion port and should have a much lower effect on aerodynamics. I also just suggested this here: https://www.bitcraze.io/2018/06/mocap-deck/.
6. In our space 250K performed worse than the 2M mode. The data rate will be a bit lower, but I think that shouldn't be a problem for the Crazyswarm.

For the cases where a drone crashes, how does the vicon.dt look like? In your plots it is hard to tell since there is a large initial spike. If you look at vicon.dt right around the crash, this should give you a good indication if there was communication loss first (causing wrong state estimation, causing loss of control), or a control issue (caused, for example, by output saturation).

**Comment by srnetsr** [comment]
Hi, is it resolved?
In my case before, when I used goTo function, then sometimes cf2 was just dropped to the ground, and rebooted by itself.
I am not sure rebooting before falling or falling before rebooting happened.
Do I need to upgrade firmware to resolve this issue?

Thank you.

**Comment by jpreiss** [comment]
@pPatrickK were you able to resolve this?

**Comment by jpreiss** [comment]
closing due to inactivity.
